### PR TITLE
feat(curriculum): add Linked list lab

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by jdwilkin4 — COMMENTED** [review]
is this the same linked list challenge found in the coding interview prep section?
if so, then we need to make sure to use the same id and add that duplicate id to the duplicate ids array like done in the other PR's that Ilenia and I have created so far

**Comment by jdwilkin4** [comment]
since this is technically a different challenge, then a new id would be appropriate here

File: curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+```
+
+Your `isEmpty` method should return `true` when there are no elements in linked list.
+
+```js
+const test = new LinkedList();
+assert.isTrue(test.isEmpty());
+```
+
+The previous node in your `LinkedList` class should have reference to the newest node created.
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md:133]:
```suggestion
The previous node in your `LinkedList` class should have a reference to the newest node created.
```

File: curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+```js
+const test = new LinkedList();
+test.add("pig");
+test.add("dog");
+test.add("sheep");
+test.add("cat");
+test.remove("elephant");
+const expected = { element: "pig", next: {element: "dog", next: {element: "sheep", next: {element: "cat", next: null}}} }
+assert.deepEqual(test.head, expected)
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md:200]:
```suggestion
assert.deepEqual(test.head, expected);
```

File: curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+```js
+const test = new LinkedList();
+test.add("pig");
+test.add("dog");
+test.add("sheep");
+test.add("cat");
+test.remove("sheep");
+const expected = { element: "pig", next: {element: "dog", next: {element: "cat", next: null}} }
+assert.deepEqual(test.head, expected)
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md:187]:
```suggestion
assert.deepEqual(test.head, expected);
```

File: curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+1. Whenever a node is removed from the linked list, the `next` property of the previous node should be set to the following node, and the length of the list should be decremented.
+
+# --hints--
+
+Your `LinkedList` class should instantiate objects with the `length` property set to `0`.
+
+```js
+const test = new LinkedList();
+assert.strictEqual(test.length, 0)
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md:38]:
```suggestion
assert.strictEqual(test.length, 0);
```

File: curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+```
+
+Your `LinkedList` class should instantiate objects with the `head` property set to `null`.
+
+```js
+const test = new LinkedList();
+assert.strictEqual(test.head, null)
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md:45]:
```suggestion
assert.strictEqual(test.head, null);
```

File: curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+```
+
+Your `LinkedList` class should have a property `Node`.
+
+```js
+const test = new LinkedList();
+assert.property(test, "Node")
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-linked-list-class/66dadcf18df3a76104054d95.md:52]:
```suggestion
assert.property(test, "Node");
```

**Review by zairahira — COMMENTED** [review]
The project is looking great.
I have left some suggestions, mostly nits.
However, I would suggest to explain what is a linked list in the description part.

**Review by jdwilkin4 — APPROVED** [review]

**Review by zairahira — APPROVED** [review]

**Pull request merged**